mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -212,12 +226,110 @@ class LinearRegressionModel private[ml] (
   extends RegressionModel[Vector, LinearRegressionModel]
   with LinearRegressionParams {
 
+  @transient private var trainingResults: Option[LinearRegressionTrainingResults] = None
+
+  /**
+   * Gets results (e.g. residuals, mse, r^2) of model on training set. This method should only
+   * be called on the driver (it is not available on workers).
+   */
+  def getTrainingResults: Option[LinearRegressionTrainingResults] = trainingResults
```

Comment:
OK, will change `trainingResults` to `summary`.

wrt results not being available and `hasSummary`, I'm currently using an `Option` to handle those two cases. Would you suggest something else?